mlflow/tracing/fluent.py

```diff
@@ -733,20 +726,29 @@ def f():
     Returns:
         The last active trace if exists, otherwise None.
     """
-    if is_in_databricks_model_serving_environment():
-        raise MlflowException(
-            "The function `mlflow.get_last_active_trace` is not supported in "
-            "Databricks model serving.",
-            error_code=BAD_REQUEST,
-        )
-
-    if len(TRACE_BUFFER) > 0:
-        last_active_request_id = list(TRACE_BUFFER.keys())[-1]
-        return TRACE_BUFFER.get(last_active_request_id)
+    trace_id = (
+        _LAST_ACTIVE_TRACE_ID_THREAD_LOCAL.get() if thread_local else _LAST_ACTIVE_TRACE_ID_GLOBAL
+    )
+    if trace_id is not None:
+        try:
+            return MlflowClient().get_trace(trace_id, display=False)
+        except:
+            _logger.debug(
+                f"Failed to get the last active trace with request ID {trace_id}.",
+                exc_info=True,
+            )
+            raise
     else:
         return None
 
 
+def _set_last_active_trace_id(trace_id: str):
+    """Internal function to set the last active trace ID."""
+    global _LAST_ACTIVE_TRACE_ID_GLOBAL
+    _LAST_ACTIVE_TRACE_ID_GLOBAL = trace_id
```

Comment:
q: should we use a lock when assigning a value to `_LAST_ACTIVE_TRACE_ID_GLOBAL`? If this is shared globally within a process, this operation should be thread-safe.

Reply:
Thanks for the clarification. I wondered what would happen if _LAST_ACTIVE_TRACE_ID_GLOBAL and _LAST_ACTIVE_TRACE_ID_THREAD_LOCAL are not the same during the call of this method, but agree this is not a problem.